src/journal-remote/journal-upload.c

```diff
@@ -121,6 +124,39 @@ static int check_cursor_updating(Uploader *u) {
         return 0;
 }
 
+int config_parse_compression(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char *compression = ASSERT_PTR(data);
+        if (isempty(rvalue)) {
+                *compression = COMPRESSION_NONE;
+                return 0;
+        }
+
+        _cleanup_free_ char *s = strdup(rvalue);
+        if (!s)
+                return log_oom();
```

Comment:
Not necessary anymore. Please drop this, and directly pass `rvalue` to `compression_from_string()`.

Also, currently all entries in `compression_table` are in upper, but please make it lower. Please do that in a different commit. Then, update docs/ENVIRONMENT.md .

Reply:
do you mean add to a table lower case values of replace existing ones with lower case?

Reply:
I think it is ok to make the existing table lower.

Reply:
This breaks backward compatibility - it's ok to support both, but the parser needs to be case-insensitive then